pkg/backends/otlp/internal/data/metric_request_test.go

```diff
@@ -0,0 +1,20 @@
+package data
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMetricsRequest(t *testing.T) {
+	t.Parallel()
+
+	req, err := NewMetricsRequest(
+		context.Background(),
+		"not-a-valid-url",
+		NewResourceMetrics(NewResource()),
+	)
+	assert.NoError(t, err, "Must not error creating request")
+	assert.NotNil(t, req, "Must have a valid request")
```

Comment:
What is your expectations with regards to checking the body? 

I wanted to avoid checking the body contents since protobuf adds extra metadata into unexposed values so it makes `assert.Equal` a pain to use.

Reply:
Just checking that the metrics that are attached to the post request as expected, at a minimum I reckon we should check the body has something